Documentation/Migrations/Hyperspace 3.x-4.0 Migration Guide.md

```diff
@@ -0,0 +1,111 @@
+                       body: try? HTTP.Body(post))
+    }
+}
+```
+
+For advanced usages, it is still possible to create unique types for each request by wrapping the framework's `Request` type and create the `Hyperspace.Request` in a computed property
+
+```swift
+struct CreatePostRequest {
+    let newPost: NewPost
+
+    var request: Request<Post, AnyError> {
+        return Request(method: .post, url: URL(string: "https://jsonplaceholder.typicode.com/posts")!, headers: [.contentType: .applicationJSON],
+                       body: try? HTTP.Body(post))
+    }
+}
+```
+
+### `NetworkService` renamed to `TransportService`
+
+Another change is the renaming of the various `Network` related types in the framework to `Transport`. This includes `Transporting` (previously `NetworkServiceProtocol`), `TransportService` (previously `NetworkService`), `TransportResult` (previously `NetworkServiceResult`), `TransportSuccess` (previously `NetworkServiceSuccess`), `TransportFailure` (previously `NetworkServiceFailure`), `TransportFailureRepresentable` (previously `NetworkServiceFailureInitializable`) and `TransportSession` (previously `NetworkSession`). In many instances, deprecated typealiases should exist to ease the transition to the new type name.
+
+### Strongly typed HTTP Headers
+
+In Hyperspace 4.0, the headers property of `Request` is now strongly typed as `public var headers: [HTTP.HeaderKey: HTTP.HeaderValue]?`. Both the key and value are easily extendable, and a wider range of presets are built in to Hyperspace.
+
+### Strongy typed HTTP Body
+
+Along with the changes to the headers, a new type has been introduced in 4.0 which abstracts over the HTTP body. Instead of accepting an arbitrary `Data` object, a new `HTTP.Body` type has been introduced to make common cases (like URL forms and JSON) easier to implement. In addition the `HTTP.Body` now supports adding additional HTTP headers that are automatically applied to any request containing that body.
+
+### Multiple Recovery Strategies
+
+The `BackendService` now supports multiple possible `RecoveryStrategy` objects (previously `RequestRecoveryStrategy`). In the cases where recovery may be applicable, these `RecoveryStrategy` will be queried in order, and the first strategy that is able to attempt a recovery operation will be used. If that recovery attempt then fails, no other strategies will be tried and the request will report a failure result back to the caller.
+
+### Error Handling
+
+There have been several changes made to the core `NetworkService` functionality to allow for more fine grained error handling. These changes include:
+
+- Any time you a custom `Error` type is initialized, the raw `HTTP.Response` received by the server will be provided. This will happen both when a transport failure and decoding failure occurs, and gives access to data like the raw http status and response data.
```

Comment:
Typo: "Any time you a custom..." --> "Any time your custom..."?

Reply:
Fixed